### DISCUSSION TITLE
New Robot not converging to a solution

### DISCUSSION THREAD

**Discussion by DaniCoimbra** [NVlabs/curobo#292] [Software Q&A]
Hello,

I have setup my own robot, from the .urdf, I exported the .usd file using isaac sim and added the collision spheres.
In Isaac Sim, all joints are moving correctly, the yml file follows the setup in the tutorial, and the collision spheres are setup respecting the "self_collision_ignore" field.

My issue: once I press start on the simulation, all I get is:
---
**** Click Play to start simulation ***** 

Updating world, reading w.r.t. /World/as1  

2  

Updated World  

2024-06-04 16:02:10 [14,710ms] [Warning] [__main__] Plan did not converge to a solution.  No action is being taken.  

2024-06-04 16:02:10 [14,979ms] [Warning] [__main__] Plan did not converge to a solution.  No action is being taken.  

---

**Comment by DaniCoimbra** [comment]
Update:
The ee_link is pointing to the middle of the end link, not the tip, how should I go about updating the end link position that should be touching the cube?

**Reply by balakumar-s** [reply]
Do you have the name of the end-effector link? ~~Is it possible for you to share the robot configuration file and the urdf?~~

Update: just saw your github repo, will try to debug your issue.

**Reply by balakumar-s** [reply]
Looking at your configuration file, make these 0.0: https://github.com/DaniCoimbra/curoboTest/blob/25a41a86fee928b6c32a92eb90c4ded7b8e6423b/as1.yml#L161

The values in this dictionary adds an offset radius to each link. Adding 5cm may cause self collision between links. I haven't tried if 0.0 fixes your problem yet, but that might solve your issue.

**Reply by DaniCoimbra** [reply]
What I understand is that the goal position is when the end-effector origin touches the goal. 
But in the cube example, how do I define which face and angle of the cube is the target for the end-effector?

**Reply by balakumar-s** [reply]
The target is the origin of the cube which is the center of mass for the cube.
